drivers/video/ov5640.c

```diff
@@ -904,7 +904,8 @@ static int ov5640_enum_frmival(const struct device *dev, enum video_endpoint_id
 		}
 	}
 
-	if (i == ARRAY_SIZE(modes) || fie->index > ARRAY_SIZE(ov5640_frame_rates) ||
+	if (i == ARRAY_SIZE(modes) || fie->index < 0 ||
+	    fie->index >= ARRAY_SIZE(ov5640_frame_rates) ||
```

Comment:
You should remove `fie->index < 0`, just do `fie->index >= ARRAY_SIZE(ov5640_frame_rates)` as `fie->index` is an unsigned value. If you compare it with 0, you will run into another coverity issue as [here](https://github.com/zephyrproject-rtos/zephyr/issues/81936)